actor-tests/src/test/scala/org/apache/pekko/io/dns/DockerBindDnsService.scala

```diff
@@ -18,89 +18,142 @@ import scala.util.Try
 import scala.util.control.NonFatal
 
 import com.typesafe.config.Config
-import com.spotify.docker.client.DefaultDockerClient
-import com.spotify.docker.client.DockerClient.{ ListContainersParam, LogsParam }
-import com.spotify.docker.client.messages.{ ContainerConfig, HostConfig, PortBinding }
+import com.github.dockerjava.core.DockerClientConfig
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.core.DefaultDockerClientConfig
+import com.github.dockerjava.api.model.Frame;
+import com.github.dockerjava.api.async.ResultCallback;
 import org.scalatest.concurrent.Eventually
 
 import org.apache.pekko
 import pekko.testkit.PekkoSpec
 import pekko.util.ccompat.JavaConverters._
+import com.github.dockerjava.api.DockerClient
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient
+import com.github.dockerjava.api.command.CreateContainerCmd
+import com.github.dockerjava.api.model.PortBinding
+import com.github.dockerjava.api.model.Container
+import com.github.dockerjava.api.model.Volume
+import com.github.dockerjava.core.DockerClientImpl
+import com.github.dockerjava.api.model.Bind
 
 abstract class DockerBindDnsService(config: Config) extends PekkoSpec(config) with Eventually {
-  val client = DefaultDockerClient.fromEnv().build()
+
+  val dockerConfig: DockerClientConfig = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+
+  val httpClient: ApacheDockerHttpClient = new ApacheDockerHttpClient.Builder()
+    .dockerHost(dockerConfig.getDockerHost())
+    .sslConfig(dockerConfig.getSSLConfig())
+    .maxConnections(100)
+    .build();
+
+  val client: DockerClient = DockerClientImpl.getInstance(dockerConfig, httpClient);
 
   val hostPort: Int
 
   var id: Option[String] = None
 
-  def dockerAvailable() = Try(client.ping()).isSuccess
+  def dockerAvailable() = Try(client.pingCmd().exec()).isSuccess
 
   override def atStartup(): Unit = {
     log.info("Running on port port {}", hostPort)
     super.atStartup()
 
     // https://github.com/sameersbn/docker-bind/pull/61
     val image = "raboof/bind:9.11.3-20180713-nochown"
+
     try {
-      client.pull(image)
+      client
+        .pullImageCmd(image)
+        .start()
     } catch {
       case NonFatal(_) =>
         log.warning(s"Failed to pull docker image [$image], is docker running?")
         return
     }
```

Comment:
Do we need to close the client in finally block？

Reply:
Adding a `finally` block here would close the connection pool of the client before executing the subsequent instruction. I proceeded by adding the `client.close()` instruction in the `afterTermination()` override